baseline/client/spatial-orientation/spatial-orientation.js

```diff
@@ -13,20 +15,60 @@ import test_instruction_html from "./frag/test_instruction.html";
 import completion_html from "./frag/completion.html";
 
 export class SpatialOrientation {
+    constructor(setNum) {
+        if (Number.isInteger(setNum) && setNum > 0) {
+            this.setNum = setNum
+        } else {
+            throw new Error("setNum must be a strictly positive integer");
+        }
+        // get test stimulus
+        const setStim = this.constructor.stimulus.pre.main[String(setNum)];
+        this.trialsStim = (
+            setStim.order === "random" ?
+            jsPsych.randomization.repeat(setStim.trials, 1) :
```

Comment:
Oh, right!
https://www.jspsych.org/core_library/jspsych-randomization/#shuffle-an-array-no-repeats
```
var myArray = [1,2,3,4,5];
var shuffledArray = jsPsych.randomization.repeat(myArray, 1);
// output: shuffledArray = [3,2,4,1,5]
```
I wonder why the example on the jsPsych reference does it like this...

Reply:
Probably because that's the reference for repeat. :) The [reference for shuffle](https://www.jspsych.org/core_library/jspsych-randomization/#shuffle-an-array) does the same thing using shuffle.

Reply:
Ah, that was silly. I <kbd>Ctrl</kbd>+<kbd>F</kbd>ed "shuffle" and [felt lucky](https://webapps.stackexchange.com/questions/31546/what-does-im-feeling-lucky-button-do).